homeassistant/components/buienradar/sensor.py

```diff
@@ -55,264 +57,582 @@
-    "barometerfcnamenl": ["Barometer", None, "mdi:gauge", None],
-    "condition": ["Condition", None, None, None],
-    "conditioncode": ["Condition code", None, None, None],
-    "conditiondetailed": ["Detailed condition", None, None, None],
-    "conditionexact": ["Full condition", None, None, None],
-    "symbol": ["Symbol", None, None, None],
+    SensorEntityDescription(
+        key="barometerfcnamenl",
+        name="Barometer",
+        icon="mdi:gauge",
+    ),
+    SensorEntityDescription(
+        key="condition",
+        name="Condition",
+    ),
+    SensorEntityDescription(
+        key="conditioncode",
+        name="Condition code",
+    ),
+    SensorEntityDescription(
+        key="conditiondetailed",
+        name="Detailed condition",
+    ),
+    SensorEntityDescription(
+        key="conditionexact",
+        name="Full condition",
+    ),
+    SensorEntityDescription(
+        key="symbol",
+        name="Symbol",
+    ),
     # new in json api (>1.0.0):
-    "feeltemperature": [
-        "Feel temperature",
-        TEMP_CELSIUS,
-        None,
-        DEVICE_CLASS_TEMPERATURE,
-    ],
-    "humidity": ["Humidity", PERCENTAGE, "mdi:water-percent", None],
-    "temperature": [
-        "Temperature",
-        TEMP_CELSIUS,
-        None,
-        DEVICE_CLASS_TEMPERATURE,
-    ],
-    "groundtemperature": [
-        "Ground temperature",
-        TEMP_CELSIUS,
-        None,
-        DEVICE_CLASS_TEMPERATURE,
-    ],
-    "windspeed": ["Wind speed", SPEED_KILOMETERS_PER_HOUR, "mdi:weather-windy", None],
-    "windforce": ["Wind force", "Bft", "mdi:weather-windy", None],
-    "winddirection": ["Wind direction", None, "mdi:compass-outline", None],
-    "windazimuth": ["Wind direction azimuth", DEGREE, "mdi:compass-outline", None],
-    "pressure": ["Pressure", PRESSURE_HPA, "mdi:gauge", None],
-    "visibility": ["Visibility", LENGTH_KILOMETERS, None, None],
-    "windgust": ["Wind gust", SPEED_KILOMETERS_PER_HOUR, "mdi:weather-windy", None],
-    "precipitation": [
-        "Precipitation",
-        PRECIPITATION_MILLIMETERS_PER_HOUR,
-        "mdi:weather-pouring",
-        None,
-    ],
-    "irradiance": [
-        "Irradiance",
-        IRRADIATION_WATTS_PER_SQUARE_METER,
-        "mdi:sunglasses",
-        None,
-    ],
-    "precipitation_forecast_average": [
-        "Precipitation forecast average",
-        PRECIPITATION_MILLIMETERS_PER_HOUR,
-        "mdi:weather-pouring",
-        None,
-    ],
-    "precipitation_forecast_total": [
-        "Precipitation forecast total",
-        LENGTH_MILLIMETERS,
-        "mdi:weather-pouring",
-        None,
-    ],
+    SensorEntityDescription(
+        key="feeltemperature",
+        name="Feel temperature",
+        unit_of_measurement=TEMP_CELSIUS,
+        device_class=DEVICE_CLASS_TEMPERATURE,
+    ),
+    SensorEntityDescription(
+        key="humidity",
+        name="Humidity",
+        unit_of_measurement=PERCENTAGE,
+        icon="mdi:water-percent",
```

Comment:
```suggestion
        device_class=DEVICE_CLASS_HUMIDITY,
```

Reply:
Technically correct, I'm just not sure this should be done here. The intent with these PRs is to limit the changes to "only" the conversion itself.

Reply:
Makes sense! :-)